crates/lib3h/src/transport/transport_multiplex/route.rs

```diff
@@ -0,0 +1,235 @@
+    fn handle_msg_from_inner(
+        &mut self,
+        mut msg: GhostMessage<
+            RequestToParent,
+            RequestToChild,
+            RequestToParentResponse,
+            TransportError,
+        >,
+    ) -> TransportResult<()> {
+        match msg.take_message().expect("exists") {
+            RequestToParent::IncomingConnection { address } => {
+                self.handle_incoming_connection(address)
+            }
+            RequestToParent::ReceivedData { address, payload } => {
+                self.handle_received_data(address, payload)
+            }
+            RequestToParent::TransportError { error } => self.handle_transport_error(error),
+        }
+    }
+
+    /// private handler for inner transport IncomingConnection events
+    fn handle_incoming_connection(&mut self, address: Url) -> TransportResult<()> {
+        // forward
+        self.endpoint_self
+            .publish(RequestToParent::IncomingConnection { address });
+        Ok(())
+    }
+
+    /// private handler for inner transport ReceivedData events
+    fn handle_received_data(&mut self, address: Url, payload: Vec<u8>) -> TransportResult<()> {
+        // forward
+        self.endpoint_self
+            .publish(RequestToParent::ReceivedData { address, payload });
+        Ok(())
+    }
+
+    /// private handler for inner transport TransportError events
+    fn handle_transport_error(&mut self, error: TransportError) -> TransportResult<()> {
+        // forward
+        self.endpoint_self
+            .publish(RequestToParent::TransportError { error });
+        Ok(())
+    }
+
+    /// private dispatcher for messages coming from our parent
+    fn handle_msg_from_parent(
+        &mut self,
+        mut msg: GhostMessage<
+            RequestToChild,
+            RequestToParent,
+            RequestToChildResponse,
+            TransportError,
+        >,
+    ) -> TransportResult<()> {
+        match msg.take_message().expect("exists") {
+            RequestToChild::Bind { spec } => self.handle_bind(msg, spec),
+            RequestToChild::SendMessage { address, payload } => {
+                self.handle_send_message(msg, address, payload)
+            }
+        }
+    }
+
+    /// private handler for Bind requests from our parent
+    fn handle_bind(
+        &mut self,
+        msg: GhostMessage<RequestToChild, RequestToParent, RequestToChildResponse, TransportError>,
+        spec: Url,
+    ) -> TransportResult<()> {
+        // forward the bind to our inner_transport
+        self.inner_transport.as_mut().request(
+            std::time::Duration::from_millis(2000),
+            RouteToInnerContext::AwaitBind(msg),
+            RequestToChild::Bind { spec },
+            Box::new(|_, context, response| {
+                let msg = {
+                    match context {
+                        RouteToInnerContext::AwaitBind(msg) => msg,
+                        _ => return Err("bad context".into()),
```

Comment:
Show enum value of bad context in error.

Reply:
done